File: ex4/2/task2.cpp
#include <iostream>
#include <string>

using namespace std;

string replace(string str) {

    string str_replaced = "";

    for (auto &c : str) {

        switch (c)
        {
        case '<':
            str_replaced += "&lt";
            break;
        case '>':
            str_replaced += "&gt";
            break;
        case '&':
            str_replaced += "&amp";
            break;
        default:
            str_replaced += c;
            break;
        }
        
    }
    
    return str_replaced;
}

int main() 
{
    cout << replace("foo <&> bar") << endl;
    cout << replace("<<<<<<&&&&&>>>>>>>") << endl;
    cout << replace("Cakes & cookies") << endl;
    cout << replace("C++ > all other languages") << endl;

    return 0;
}

File: ex4/2/task2.rs
fn main() {
    let test1 = String::from("foo <&> bar");
    let test2 = String::from("<<<<<<&&&&&>>>>>>>");
    let test3 = String::from("Cakes & cookies");
    let test4 = String::from("Rust > all other languages");
    
    println!("{}", replace(test1));
    println!("{}", replace(test2));
    println!("{}", replace(test3));
    println!("{}", replace(test4));
}

fn replace (str: String) -> String {
    let mut str_replacement = str.clone();
    str_replacement = str_replacement
        .replace("&", "&amp")
        .replace("<", "&lt")
        .replace(">", "&gt");
    
    return str_replacement;
}